agentflow/agent_flow.py

```
@@ -0,0 +1,1138 @@
+            # Archive case to disk before eviction
+            self._archive_case(cid)
+            del self.cases[cid]
+        print(f"[Memory] Evicted {evict_count} old cases to stay under {self.max_cases} case limit")
+
+    def _archive_case(self, case_id: str):
+        """Archive a case to disk before eviction."""
+        import json
+        case = self.cases.get(case_id)
+        if not case:
+            return
+        archive_dir = os.path.join(os.path.dirname(__file__), "data", "archived_cases")
```

Ruff (0.15.7)

[error] Undefined name `os`

(F821)